src/Codata/Colist.agda

```diff
@@ -66,27 +67,31 @@ module _ {ℓ} {A : Set ℓ} where
 
 module _ {a b} {A : Set a} {B : Set b} where
 
- map : ∀ (f : A → B) → ∀ {i} → Colist A i → Colist B i
+ map : ∀ {i} (f : A → B) → Colist A i → Colist B i
```

Comment:
Is there a reason you're moving all these implicits earlier?

Reply:
It allowed me to use these functions directly in the `.Categorical` module. Otherwise I would have
needed to eta-expand them because e.g. `(A -> B) -> Colist A i -> Colist B i` does not unify
with `(A -> B) -> forall {i} -> Colist A i -> Colist B i`.

Ultimately I think I realized / came to believe that there was no advantage whatsoever in interleaving
explicit and implicit quantifiers and decided to lift all the implicit ones to a prenex position.

I do not mind putting them back where they were if necessary.